__tests__/app.test.js

```diff
@@ -29,62 +30,116 @@ describe('app routes', () => {
       return client.end(done);
     });
 
+    test('GET /characters returns list of characters', async() => {
+
+      const expectation = characterData.map(character => character.name);
+      const expectedData = {
+        id: 1,
+        name: 'Simon Kaine',
+        bad: false,
+        species_id: 1
+      };
+
+      const data = await fakeRequest(app)
+        .get('/characters')
+        .expect('Content-Type', /json/)
+        .expect(200);
+
+      const name = data.body.map(character => character.name);
+
+      expect(name).toEqual(expectation);
+      expect(name.length).toBe(characterData.length);
+      expect(data.body[0]).toEqual(expectedData);
+    }, 10000);
+
+    test('GET /characters/:id returns the individual character', async ()=>{
+      const expectation = {
+        id: 1,
+        name: 'Simon Kaine',
+        bad: false,
+        species_id: 1
+      };
+      
+      const data = await fakeRequest(app)
+        .get('/characters/1')
+        .expect('Content-Type', /json/)
+        .expect(200);
+
+      
+      expect(data.body).toEqual(expectation);
+    });
+
+    test('GET /breeds returns list of breeds', async() => {
+      const expected = breedsData.map(breed => breed.name);
+      const data = await fakeRequest(app)
+        .get('/breeds')
+        .expect('Content-Type', /json/)
+        .expect(200);
+        
+      const speciesNames = data.body.map(breed => breed.name);
+        
+      expect(speciesNames).toEqual(expected);
+      expect(speciesNames.length).toBe(speciesNames.length);
+      expect(data.body[0].id).toBeGreaterThan(0);
+    });
+
     test('POST /characters creates a new character', async ()=>{
 
       const newCharacter = {
         id:6,
         name: 'JOY',
         bad: false,
-        species: 'Pokemon'
+        species_id: 3
       };
 
       const data = await fakeRequest(app)
         .post('/characters')
         .send(newCharacter)
         .expect(200)
+        // console.log(data.body)
         .expect('Content-Type', /json/);
       expect(data.body.name).toEqual(newCharacter.name);
       expect(data.body.id).toBeGreaterThan(0);
 
     });
 
-    test('PUT /characters/:id updates characters', async ()=>{
-      const characterUPDATEdata =   {
-        id: 1,
-        name:'Simon Kaine',
+    test('POST /characters/:id create new character into array.', async ()=>{
```

Comment:
silly naming thing but you're not just adding the character to an array -- you're adding it to the database